File: backend/BoundfoxStudios.Smudgy.Data/Models/Runtime/Session.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace BoundfoxStudios.Smudgy.Data.Models.Runtime
{
  public record Session
  {
    public Guid Id { get; set; }
    public Guid HostPlayerId { get; set; }
    public SessionConfiguration Configuration { get; set; }
    public SessionState State { get; set; }
    public ICollection<SessionRound> Rounds { get; } = new Collection<SessionRound>();
    public IDictionary<Guid, SessionPlayerInformation> Players { get; } = new Dictionary<Guid, SessionPlayerInformation>();
  }

  public record SessionPlayerInformation
  {
    public int Points { get; set; }
  }

  public record SessionRound
  {
    public Guid DrawerPlayerId { get; set; }
    public string WordToDraw { get; set; }
    public ICollection<Guid> CorrectGuesserPlayerIds { get; } = new Collection<Guid>();
  }

  public enum SessionLanguage
  {
    German,
    English,
  }

  public enum SessionState
  {
    Lobby
  }

  public record SessionConfiguration
  {
    public SessionLanguage Language { get; set; }
    public int RoundTimeInSeconds { get; set; }
    public int RoundsToPlay { get; set; }
    public int MaxPlayers { get; set; }
  }
}

File: backend_oldold/src/models/session.ts
import { SessionConfiguration } from './shared/session-configuration';

export enum SessionState {
  Lobby,
  Playing
}

export interface Round {
  drawerPlayerId: string;
  wordToDraw: string;
  correctGuesserPlayerIds: string[];
}

export interface SessionPlayer {
  playerId: string;
  points: number;
}

export interface SessionPlayers {
  [key: string]: SessionPlayer;
}

export interface Session {
  id: string;
  hostPlayerId: string;
  configuration: SessionConfiguration;
  playerIds: SessionPlayers;
  rounds: Round[];
  state: SessionState,
}

export const createSession = (id: string, hostPlayerId: string, configuration: SessionConfiguration): Session => {
  return {
    id,
    hostPlayerId,
    playerIds: {},
    configuration,
    rounds: [],
    state: SessionState.Lobby,
  };
};
